packages/doenetml-worker-rust/lib-doenetml-core/src/core/components/doenet/text_input/state_vars/immediate_value.rs

```diff
@@ -4,93 +4,90 @@ use super::TextInputState;
 
 /// The dependencies of the immediate_value state variable of the text input component
 #[add_dependency_data]
-#[derive(Debug, Default, StateVariableDependencies, StateVariableDependencyInstructions)]
-struct ImmediateValueDependencies {
-    essential: StateVarReadOnlyView<String>,
-    sync_immediate_value: StateVarReadOnlyView<bool>,
-    bind_value_to: StateVarReadOnlyView<String>,
-    prefill: StateVarReadOnlyView<String>,
+#[derive(Debug, Default, StateVariableDependencies, StateVariableDataQueries)]
+struct RequiredData {
+    essential: StateVarView<String>,
+    sync_immediate_value: StateVarView<bool>,
+    bind_value_to: StateVarView<String>,
+    prefill: StateVarView<String>,
 }
 
 /// The interface for the immediate_value state variable of a text input
 #[derive(Debug, Default)]
-pub struct ImmediateValueStateVarInterface {
-    /// The dependency instructions that indicate how the dependencies of this state variable will be created.
-    dependency_instructions: ImmediateValueDependencyInstructions,
+pub struct ImmediateValueStateVar {
+    /// The graph queries that indicate how the dependencies of this state variable will be created.
+    data_queries: RequiredDataDataQueries,
```

Comment:
It seems like "graph query" and "data query" are synonyms.

Reply:
I'm trying to figure out if we are trying to eliminate the term "dependencies" in favor of the term "data". They have significantly different connotations. Dependencies emphasizes the graph edges; data emphasizes the result that you need to do your calculations.

It make sense to rename "dependency values" as data.

Reply:
Yeah, I feel like the more we can isolate things, the better. Core certainly thinks about graphs, but components should just be handed what they need and work with that without caring where things come from.

Reply:
Docstrings fixed.